feat: Added support for MSafe Wallet

jovicheng opened the pull request:
feat: Added support for MSafe Wallet

jovicheng committed:
feat: Added support for MSafe Wallet

0xmaayan commented on .changeset/late-cities-dress.md:
```diff
@@ -0,0 +1,5 @@
+---
+"@aptos-labs/wallet-adapter-core": major
```
This should not be a major update, please change to minor

0xmaayan commented on packages/wallet-adapter-core/src/WalletCore.ts:
```diff
@@ -1050,13 +1054,15 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
 
       const aptosConfig = getAptosConfig(this._network, this._dappConfig);
-      const signingMessage = new TextEncoder().encode(response.args.fullMessage);
+      const signingMessage = new TextEncoder().encode(
```
Please revert formatting

0xmaayan commented on packages/wallet-adapter-core/src/sdkWallets.ts:
```diff
@@ -50,6 +51,15 @@ export function getSDKWallets(dappConfig?: DappConfig) {
     sdkWallets.push(new DevTWallet() as any);
   }
 
+  if (dappConfig?.network) {
```
Whats the reason for this check? It doesn't look like it is a required prop for mSafe

jovicheng replied:
<img width="253" alt="image" src="https://github.com/user-attachments/assets/4a9b75b8-9101-45de-8968-f228443e55af" />
`network` is required prop for MSafeConfig

jovicheng committed:
fix: Update wallet adapter core version to minor and improve network handling

- Changed version of "@aptos-labs/wallet-adapter-core" from major to minor
- Improved network handling in MSafeWallet configuration
- Refactored signing message encoding for clarity
